packages/protocol/lib/compatibility/ast-version.ts

```diff
@@ -54,9 +57,9 @@ export async function getContractVersion(artifact: Artifact): Promise<ContractVe
 }
 
 export class ASTContractVersionsChecker {
-  static create = async (oldArtifacts: BuildArtifacts, newArtifacts: BuildArtifacts, expectedVersionDeltas: ContractVersionDeltaIndex): Promise<ASTContractVersionsChecker> => {
-    const oldVersions = await ASTContractVersions.fromArtifacts(oldArtifacts)
-    const newVersions = await ASTContractVersions.fromArtifacts(newArtifacts)
+  static create = async (oldArtifacts: BuildArtifacts, newArtifactsSet: BuildArtifacts[], expectedVersionDeltas: ContractVersionDeltaIndex): Promise<ASTContractVersionsChecker> => {
+    const oldVersions = await ASTContractVersions.fromArtifacts([oldArtifacts])
```

Comment:
Do we want to mix old artifacts for 0.5 and 0.8 ?

Reply:
Yes, it's possible to check versions with different solidity versions.